File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.py
def solution(participant, completion):
    answer = ''
    dic = {}
    for par in participant:
        if par in dic:
            dic[par]+=1
        else:
            dic[par] = 1
    for complete in completion:
        dic[complete] -=1
    itemList = list(dic.items())
    return list(filter(lambda x:x[1] == 1, itemList))[0][0]

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.*;
import java.util.Map.*;
class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String, Integer> map = new HashMap<>();
        for(int i = 0 ; i<participant.length; i++){
            Integer count = map.getOrDefault(participant[i], 0);
            map.put(participant[i], count + 1);
        }
        for(int i = 0 ; i<completion.length; i++){
            Integer count = map.getOrDefault(completion[i], 0);
            map.put(completion[i], count - 1);
        }
        for(Entry entry: map.entrySet()){
            if((Integer)entry.getValue() == 1){
                return (String)entry.getKey();
            }
        }
        return "";
    }
}